### PR TITLE
Add node browser polyfills for process and Buffer

### LINKED ISSUE #2662: [6.x] Legacy Node Polyfills
We should either add a config option / component that re-enables these or a note to the docs.

What needs to happen is rather simple:

1. Install `node-libs-browser`
2. Update the webpack config
```js
mix.webpackConfig({
    resolve: {
        fallback: {
            // This handles all of them…
            // ...Object.fromEntries(Object.entries(require("node-libs-browser")).filter(e => e[1] !== null)),

            // … but these are likely the most used and as such should be sufficient
            process: require("node-libs-browser").process,
            buffer: require("node-libs-browser").buffer,
        },
    },
})
```

If we were to add a shortcut for this in Mix I'd probably do something like this:
```js
mix.options({ legacyNodePolyfills: true })
```

What do you think @JeffreyWay? Would documentation suffice or should we add a shortcut for this?

### LINKED ISSUE #2685: process.ENV blank in v6
Previous versions when we had a .env file we could do things such as...

MIX_API_KEY=123456

Then access it via JavaScript using 

process.env.MIX_API_KEY

however this is now coming up as undefined, I'm assuming this is because webpack removed it by default? https://webpack.js.org/migrate/5/ says to use DefinePlugin or EnvironmentPlugin

Could we bundle that in for compatibility?

### PR DESCRIPTION
**Pull request by thecrypticace**
Fixes #2662
Fixes #2685

### CODE REVIEW

File: src/components/LegacyNodePolyfills.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        return [
+            new webpack.ProvidePlugin({
+                Buffer: 'buffer',
+                process: 'process/browser'
+            })
+        ];
+    }
+
+    webpackConfig() {
+        return {
```
Comment (JeffreyWay) [inline src/components/LegacyNodePolyfills.js:19]: This will be added regardless of if `Config.legacyNodePolyfills` is set. Is that okay? Any issue there?

Comment (thecrypticace) [reply src/components/LegacyNodePolyfills.js:19]: Heh oops. Fix incoming

**Comment by JeffreyWay** [comment]
Perfect. Thanks!

**Comment by soilSpoon** [comment]
`node-libs-browser` is deprecated. so i think using this library could hurt UX.
Therefore i think it should change from webpack/node-libs-browser to feross/buffer and shtylman/node-process.

What do you think about the above?

**Comment by thecrypticace** [comment]
Yeah we should probably do that. `node-libs-browser` is, in reality, a package that serves only to include other packages + provide a unified way to get what import is useful for the browser. I missed the note that it is deprecated.

**Comment by Kaos-Industries** [comment]
I can only get this solution working if I do the following:

```
mix
	.js("resources/js/app.js", "public/js")
	.js("resources/js/payments.js", "public/js")      // Had to add this 
	.sass("resources/sass/app.scss", "public/css")
	.postCss("resources/css/app.css", "public/css")
	.sourceMaps()
	.options({
		legacyNodePolyfills: true,      // And this
	});
```

...where `payments.js` is the file I'm trying to access my Laravel environment variables from. This isn't ideal because, at least without npm's `--production` flag, the `payments.js` file gets "compiled" to a size that's about ten times bigger than the pre-compiled file. Is there a way to get this fix working without that extra .js compilation line in my `webpack.mix.js`?

**Pull request merged**